sklearn/ensemble/_hist_gradient_boosting/tests/test_compare_lightgbm.py

```diff
@@ -77,14 +96,21 @@ def test_same_predictions_regression(seed, min_samples_leaf, n_samples, max_leaf
 
     pred_lightgbm = est_lightgbm.predict(X_train)
     pred_sklearn = est_sklearn.predict(X_train)
-    # less than 1% of the predictions are different up to the 3rd decimal
-    assert np.mean(abs(pred_lightgbm - pred_sklearn) > 1e-3) < 0.011
-
-    if max_leaf_nodes < 10 and n_samples >= 1000:
+    if loss in ("gamma", "poisson"):
+        # More than 65% of the predictions must be close up to the 2nd decimal.
+        assert (
+            np.mean(np.isclose(pred_lightgbm, pred_sklearn, rtol=1e-2, atol=1e-2))
+            > 0.65
+        )
+    else:
+        # Less than 1% of the predictions may deviate more than 1e-3 in relative terms.
+        assert np.mean(np.isclose(pred_lightgbm, pred_sklearn, rtol=1e-3)) > 1 - 0.01
+
+    if max_leaf_nodes < 10 and n_samples >= 1000 and loss in ("squared_error"):
```

Comment:
I don't understand how the current code can possibly work. It think the condition was never met and the assertion skipped:

```suggestion
    if max_leaf_nodes < 10 and n_samples >= 1000 and loss in ("squared_error",):
```

Reply:
The pair `n_samples, max_leaf_nodes = 1000, 8` triggers this test.